eppoclient/evalbandits.go

```diff
@@ -0,0 +1,234 @@
+package eppoclient
+
+import (
+	"math"
+	"sort"
+	"strconv"
+)
+
+type ContextAttributes struct {
+	Numeric     map[string]float64
+	Categorical map[string]string
+}
+
+// Tries to map generic attributes to ContextAttributes depending on attribute types.
+// - Integer and float types are mapped to numeric attributes.
+// - Strings and bools are mapped to categorical attributes.
+// - Rest of types are silently dropped.
+func InferContextAttributes(attrs map[string]interface{}) ContextAttributes {
+	result := ContextAttributes{
+		Numeric:     map[string]float64{},
+		Categorical: map[string]string{},
+	}
+	for key, value := range attrs {
+		switch value := value.(type) {
+		case int:
+			result.Numeric[key] = float64(value)
+		case int8:
+			result.Numeric[key] = float64(value)
+		case int16:
+			result.Numeric[key] = float64(value)
+		case int32:
+			result.Numeric[key] = float64(value)
+		case int64:
+			result.Numeric[key] = float64(value)
+		case uint:
+			result.Numeric[key] = float64(value)
+		case uint8:
+			result.Numeric[key] = float64(value)
+		case uint16:
+			result.Numeric[key] = float64(value)
+		case uint32:
+			result.Numeric[key] = float64(value)
+		case uint64:
+			result.Numeric[key] = float64(value)
+		case float32:
+			result.Numeric[key] = float64(value)
+		case float64:
+			result.Numeric[key] = value
+		case string:
+			result.Categorical[key] = value
+		case bool:
+			result.Categorical[key] = strconv.FormatBool(value)
+		}
```

Comment:
We're a lof of the same thing here. Would it be possible to group everything together so we just have one call to `float64(value)`?

Reply:
Unfortunately, no. When calling `float64()`, Go has to know the type of argument and the only way to do that is to single out every possible option